### ISSUE TITLE
"openssl list-cipher-commands" does not work with OpenSSL 1.1.0e; should be "openssl list -cipher-commands"

### ISSUE THREAD

**Issue by ircecho** [elasticdog/transcrypt#39]

**Comment by ircecho** [comment]
This problem is only present in the later OpenSSL versions, mine is 

```
OpenSSL 1.1.0e  16 Feb 2017
```

**Comment by dfee** [comment]
Specifically, here is the update that needs to be made:

```bash
validate_cipher() {
	local supported=$(openssl list -cipher-commands | tr -s ' ' '\n' | grep --line-regexp "$cipher")
	if [[ ! $supported ]]; then
		if [[ $interactive ]]; then
			printf '"%s" is not a valid cipher; choose one of the following:\n\n' "$cipher"
			openssl list -cipher-commands | column -c 80
			printf '\n'
			unset cipher
		else
			die 1 '"%s" is not a valid cipher; see `openssl list -cipher-commands`' "$cipher"
		fi
	fi
}
```

There are two changes:

1. `openssl list -cipher-commands` as mentioned above
2. `openssl list -cipher-commands | tr -s ' ' '\n' | grep --line-regexp "$cipher"` note the `tr -s` to turn the tabular data into a single column list.

If my bash-fu were better, I'd issue a PR, but I can't quite figure out how to do a conditional on the semver of openssl. i.e. `openssl version | awk '{print $2}'` returns the version... and we know that 1.1 is where the change broke, but I'm not quite there in bash land for parsing major / minor / patch and doing the comparison. Hopefully this is a good starting point for the author.

For now, I'm just adding it to my repo's /scripts dir.

**Comment by jackrh** [comment]
This issue popped up for me when I tried to decrypt a new cloned repository using the provided transcrypt command from the original repo. Applying dfee's validate_cipher update fixed it for me.

**Comment by cdimitroulas** [comment]
I have made a PR from @thomas-hilaire's fork which has the fix for this - not sure why he didn't make it himself.

**Comment by elasticdog** [comment]
This has been resolved by #48 and a7e8aef. I'll be cutting a new release shortly, but feel free to try out the code in the _master_ branch.
